hoomd/GSDDequeWriter.cc

```diff
@@ -59,14 +59,42 @@ void GSDDequeWriter::analyze(uint64_t timestep)
         }
     }
 
-void GSDDequeWriter::dump()
+void GSDDequeWriter::dump(long int start, long int end, bool clear_entire_buffer)
     {
-    for (auto i {static_cast<long int>(m_frame_queue.size()) - 1}; i >= 0; --i)
+    auto buffer_length = static_cast<long int>(m_frame_queue.size());
+    if (end > buffer_length)
+        {
+        throw std::runtime_error("Burst.dump's end index is out of range.");
+        }
+    if (start < 0 || start > buffer_length)
+        {
+        throw std::runtime_error("Burst.dump's start index is out of range.");
+        }
+    long int iterator_start, iterator_end;
+    if (end < 0)
+        {
+        iterator_end = buffer_length - start;
+        iterator_start = 0;
+        }
+    else
+        {
+        iterator_end = buffer_length - start;
+        iterator_start = buffer_length - end;
+        }
```

Comment:
So if `end < 0`, we write out the range `0, length-start`? Why is the correct range not `start, length+end`?

Reply:
Anything beyond buffer_length would cause out of bounds accessing problems. 
The reason why this is weird is because the dequeue here is in reverse order. So to dump the frames in their time progression we have to do this weird index swapping stuff.